pkg/apis/core/validation/validation_test.go

```diff
@@ -25480,6 +25475,24 @@ func TestValidatePodResize(t *testing.T) {
 			),
 		)...)
 	}
+	mkPodWithInitCtrs := func(req, lim core.ResourceList, tweaks ...podtest.TweakContainer) *core.Pod {
+		tweaks = append(tweaks, podtest.SetContainerRestartPolicy(containerRestartPolicyAlways))
+		return podtest.MakePod("pod",
+			podtest.SetInitContainers(
+				podtest.MakeContainer(
+					"restartable-init",
+					append(tweaks,
+						podtest.SetContainerResources(
+							core.ResourceRequirements{
+								Requests: req,
+								Limits:   lim,
+							},
+						),
+					)...,
+				),
+			),
+		)
+	}
```

Comment:
You can reuse `mkPodWithInitContainers` below. It also has `restartPolicy` field.